src/natcap/invest/coastal_blue_carbon/coastal_blue_carbon.py

```diff
@@ -188,941 +187,1485 @@
+            paths representing the magnitude (0-1, float32) of a disturbance.
+        args['half_life_rasters'] (dict): A 2-level deep dict structure mapping
+            int transition years to string pools to raster paths representing
+            the half-life of the given carbon pool at that transition year.
+        args['annual_rate_of_accumulation_rasters'] (dict): A 2-level deep dict
+            structure mapping int transition years to string pools to raster
+            paths representing the annual rate of accumulation for this pool in
+            the given transition period.
+        args['carbon_prices_per_year'] (dict): A dict mapping int years to the
+            floating-point price of carbon in that year.  Every year between
+            the baseline year and the analysis year (inclusive) must have a
+            key-value pair defined.
+        args['discount_rate'] (number): The discount rate for carbon
+            valuation.
+        args['analysis_year'] (int): The analysis year.  Must be greater than
+            or equal to the final transition year.
+        args['do_economic_analysis'] (bool): Whether to do valuation.
+        args['baseline_lulc_raster'] (string): The path to the baseline lulc
+            raster on disk.
+        args['baseline_lulc_year'] (int): The year of the baseline scenario.
+        args['sequestration_since_baseline_raster'] (string): The string path
+            to a raster on disk representing the total carbon sequestration
+            across all 3 carbon pools in the entire baseline period.
+        args['stocks_at_first_transition'] (dict): A dict mapping pool strings
+            (see above for the valid pool identifiers) to rasters representing
+            the carbon stocks at the end of the baseline period.
 
     Returns:
-        is_transition_year (bool): whether the year corresponding to the
-            timestep is a transition year.
+        ``None``.
+
     """
-    if (timestep_to_transition_idx(snapshot_years, transitions, timestep) !=
-        timestep_to_transition_idx(snapshot_years, transitions, timestep-1) and
-            timestep_to_transition_idx(snapshot_years, transitions, timestep)):
-        return True
-    return False
+    try:
+        n_workers = int(args['n_workers'])
+    except (KeyError, ValueError, TypeError):
+        # KeyError when n_workers is not present in args
+        # ValueError when n_workers is an empty string.
+        # TypeError when n_workers is None.
+        n_workers = -1  # Synchronous mode.
+
+    taskgraph_cache_dir = os.path.join(
+        args['workspace_dir'], TASKGRAPH_CACHE_DIR_NAME)
+    task_graph = taskgraph.TaskGraph(
+        taskgraph_cache_dir, n_workers)
+
+    suffix = args.get('suffix', '')
+    intermediate_dir = os.path.join(
+        args['workspace_dir'], INTERMEDIATE_DIR_NAME)
+    output_dir = os.path.join(
+        args['workspace_dir'], OUTPUT_DIR_NAME)
```

Comment:
Yeah, I was on the fence about having the setup be in a single function, since it's often clearer to have the setup clearly stated where it's needed rather than hidden away in a function.  In this case, the benefit clearly outweighs the cost of having this in a helper function, and the setup is more consistent as a result.  Patched in 6517b782